pkgs/tools/filesystems/xfsdump/default.nix

```diff
@@ -33,7 +33,9 @@ stdenv.mkDerivation rec {
     ncurses
   ];
 
-  patchPhase = ''
+  patches = [ ./remove-dmapapi.patch ];
```

Comment:
please add a comment saying why this patch is necessary and when it can be removed. basically "fixes build against xfsprogs >= 5.18, can probably be removed on the next update, taken from https://lore.kernel.org/linux-xfs/20220203174540.GT8313@magnolia/"